EIPS/eip-777.md

```diff
@@ -0,0 +1,551 @@
+```
+
+Notify the transmission of `amount` of tokens from the `from` address.
+
+> **parameters**
+> - `operator`: address which triggered the transfer, either sender for a direct send or an authorized operator for `operatorSend`
+> - `from`: token holder (sender)
+> - `to`: tokens recipient (or `0x` for burning)
+> - `amount`: number of tokens sent or burned
+> - `userData`: information attached to the transaction by the sender
+> - `operatorData`: information attached to the transaction by the operator
+
+###### Burning Versus Sending
+
+When tokens are sent as a result of burning:
+ - `to` MUST be `0x0`
+ - `userData` MUST be empty
+ - `operator` MUST be the address which initiated the burning
+ - `operatorData` MAY contain data
+
+When tokens are sent as a result of sending (`send` or `operatorSend`):
+ - `to` MUST be the address to which the tokens will be sent
+ - `to` MUST NOT be `0x0`
+
+ If it is a direct `send` (i.e. not an `operatorSend`) the `operator` MUST be the address from which the tokens originate. That is the `from` and `operator` addresses MUST be equals.
+
+
+### ERC777TokensRecipient
+
+Any address (contract or regular account) CAN register a contract (itself or an other) implementing the `ERC777TokensRecipient` interface via the ERC820 registry.
+
+``` ts
+interface ERC777TokensRecipient {
+    function tokensReceived(address operator, address from, address to, uint amount, bytes userData, bytes operatorData) public;
+}
+```
+#### Methods
+
+##### tokensReceived
+
+``` ts
+function tokensReceived(address operator, address from, address to, uint amount, bytes userData, bytes operatorData) public
+```
+
+Notify the transmission of `amount` of tokens to the `to` address.
+
+> **parameters**
+> - `operator`: address which triggered the transfer, either sender for a direct send or an authorized operator for `operatorSend`
+> - `from`: token holder (sender or `0x` for minting)
+> - `to`: tokens recipient (or `0x` for burning)
+> - `amount`: number of tokens sent, minted or burned
+> - `userData`: information attached to the transaction by the sender
+> - `operatorData`: information attached to the transaction by the operator
+
+###### Minting Versus Sending
+
+When tokens are received as a result of minting:
+ - `from` MUST be `0x0`
+ - `userData` MUST be empty
+ - `operator` MUST be the address which initiated the minting
+ - `operatorData` MAY contain data.
+
+When tokens are received as a result of sending (`send` or `operatorSend`):
+ - `from` MUST be the one from which the tokens originate and
+ - `from`  MUST NOT be `0x0`.
+
+If it is a direct `send` (i.e. not an `operatorSend`) the `operator` MUST be the address from which the tokens originate. That is the `from` and `operator` addresses MUST be equals.
+
+### Logo
+<table>
+  <tr>
+    <th>Image</th>
+    <th><img src="https://github.com/ethereum/EIPs/blob/master/EIPs/eip-777/logo/png/ERC-777-logo-beige-192px.png?raw=true" height="46px" align="top"></img></th>
```

Comment:
Change this to use markdown rather than HTML and use relative links instead of absolute links.  This way the images work both in the PR and on any branch.

I think the syntax is something like this for the image link, though you should look at the GitHub Markdown docs for specifics.
```
![Beige](EIPs/eip-777/logo/png/ERC-777-logo-beige-192px.png)
```